public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -125,17 +125,18 @@ export class DashboardModel implements TimeModel {
     lastRefresh: true,
   };
 
-  constructor(data: any, meta?: DashboardMeta, private getVariablesFromState: GetVariables = getVariablesByKey) {
+  constructor(data: Dashboard, meta?: DashboardMeta, private getVariablesFromState: GetVariables = getVariablesByKey) {
```

Comment:
```suggestion
  constructor(data: Partial<Dashboard>, meta?: DashboardMeta, private getVariablesFromState: GetVariables = getVariablesByKey) {    
```

Can't we change this constructor to take in a partial Dashboard model (like before)? Then we don't need all these calls to  createDashboardModelFixture({}) that to me obscures a bit what is going on. 

@joshhunt @ivanortegaalba

Reply:
What is the type that comes over the wire? `Dashboard` or `Partial<Dashboard>`?

Reply:
@joshhunt Partial, as this also used in import where this ctor accept any old version, and them adds defaults + migration

Reply:
Yeah, that's the part im unsure about - old versions are not simply partial, but a completely different type The point is that migration takes a completely incompatible type and turns it into a Dashboard.

Reply:
because the migration and defaults are applied inside the constructor (and not in the backend), it does accept anything (even an empty object).

Reply:
So unless we start applying defaults and running migrations outside the constructor the type of that param is never really "correct", Partial made it a lot easier to use from unit tests etc, and forced it to handle dashboard json models with missing values